src/components/Compiler/Compiler.ts

```diff
@@ -10,7 +10,11 @@ import { BaseCompiler } from "./base/BaseCompiler";
 export class Compiler extends BaseCompiler {
     public exports: ExportModel[];
 
-    public constructor(outputDir: string, public logicalTypes?: { [key: string]: string }) {
+    public constructor(
+        outputDir: string,
+        public logicalTypes?: { [key: string]: string},
+        public transformName?: (input: string) => string,
```

Comment:
I think it would be better if the second param is an `opts` object like so:

```ts
outputDir: string,
{
  logicalTypes?: { [key: string]: string },
  transformName?: (input: string) => string
}
```

which keeps it extensible. I came across a few other things that should be configurable and we might want to add in the future

Reply:
makes sense, i'll make it a config object.